core/src/test/java/com/linecorp/armeria/client/observation/ObservationClientIntegrationTest.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.observation;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+import org.junit.AfterClass;
+import org.junit.AssumptionViolatedException;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.client.ClientFactory;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.util.SafeCloseable;
+import com.linecorp.armeria.internal.common.observation.MicrometerObservationRegistryUtils;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.ThreadLocalCurrentTraceContext;
+import brave.test.http.ITHttpAsyncClient;
+import io.micrometer.common.KeyValues;
+import io.micrometer.observation.ObservationRegistry;
+import okhttp3.Protocol;
+
+@RunWith(Parameterized.class)
+public class ObservationClientIntegrationTest extends ITHttpAsyncClient<WebClient> {
+
+    @Parameters
+    public static List<SessionProtocol> sessionProtocols() {
+        return ImmutableList.of(SessionProtocol.H1C, SessionProtocol.H2C);
+    }
+
+    /**
+     * OkHttp's MockWebServer does not support H2C with HTTP/1 upgrade request.
+     */
+    private static final ClientFactory clientFactoryWithoutUpgradeRequest =
+            ClientFactory.builder().useHttp2Preface(true).build();
```

Comment:
`useHttp2Preface` is `true` by default so I think we can remove this `clientFactoryWithoutUpgradeRequest` completely.